### PR TITLE
0/0-Error in climatology + minor flaws in mapping

### PR DESCRIPTION
**Pull request by BenMGeo**
Found mask errors in climatology (number = sum/mean => 0/0 = nan => masked out).
Resolved by using aggregated self.data.mask (max number of pixels).

Additionally: found and resolved return_object issues in higher level plots (basemap, cartopy)  in mapping.

### CODE REVIEW

File: geoval/core/data.py [hunk 1]
```diff
@@ -350,7 +344,7 @@ def date2num(self, t):
         if self.time_str is None:
             raise ValueError('date2num can not work without timestr!')
         else:
-            print self.calendar
+            #print self.calendar
```
[the remark this answers is not in the document]
Comment (bulli92) [reply geoval/core/data.py:347]: either remove the comment totally or comment as '# '

File: geoval/core/data.py [hunk 2]
```diff
@@ -1524,7 +1518,7 @@ def _get_default_name(F, defaults):
 
 
 
-    def get_zonal_mean(self, return_object=False):
+    def get_zonal_mean(self, return_object=True):
```
[the remark this answers is not in the document]
Comment (bulli92) [reply geoval/core/data.py:1521]: only 1 empty line; delete remaining ones

File: geoval/core/data.py [hunk 3]
```diff
@@ -1940,7 +1936,7 @@ def timmean(self, return_object=False):
             return res
 
 
-    def timvar(self, return_object=False):
+    def timvar(self, return_object=True):
```
[the remark this answers is not in the document]
Comment (bulli92) [reply geoval/core/data.py:1939]: only 1 empty line; delete remaining ones

File: geoval/core/data.py [hunk 4]
```diff
@@ -1999,7 +1995,7 @@ def timsum(self, return_object=False):
             return res
 
 
-    def timn(self, return_object=False):
+    def timn(self, return_object=True):
```
[the remark this answers is not in the document]
Comment (bulli92) [reply geoval/core/data.py:1998]: only 1 empty line; delete remaining ones

File: geoval/core/data.py [hunk 5]
```diff
@@ -2009,7 +2005,7 @@ def timn(self, return_object=False):
         return_object : bool
             return Data object
         """
-        res = self.timsum() / self.timmean()
+        res = self.timsum(return_object=False) / self.timmean(return_object=False)
```
[the remark this answers is not in the document]
Comment (bulli92) [reply geoval/core/data.py:2008]: ignore this one

File: geoval/core/data.py [hunk 6]
```diff
@@ -2022,7 +2018,7 @@ def timn(self, return_object=False):
             return res
 
 
-    def timstd(self, return_object=False):
+    def timstd(self, return_object=True):
```
[the remark this answers is not in the document]
Comment (bulli92) [reply geoval/core/data.py:2021]: only 1 empty line; delete remaining ones

File: geoval/core/data.py [hunk 7]
```diff
@@ -2054,7 +2050,7 @@ def timstd(self, return_object=False):
             return res
 
 
-    def timmin(self, return_object=False):
+    def timmin(self, return_object=True):
```
[the remark this answers is not in the document]
Comment (bulli92) [reply geoval/core/data.py:2053]: only 1 empty line; delete remaining ones

File: geoval/core/data.py [hunk 8]
```diff
@@ -2142,7 +2138,7 @@ def timcv(self, return_object=True):
             return res
 
 
-    def timsort(self, return_object=False):
+    def timsort(self, return_object=True):
```
[the remark this answers is not in the document]
Comment (bulli92) [reply geoval/core/data.py:2141]: only 1 empty line; delete remaining ones

File: geoval/core/data.py [hunk 9]
```diff
@@ -2436,7 +2434,7 @@ def fldmean(self, return_data=False, apply_weights=True):
                 x[:, :] = tmp[0]
             else:
                 raise ValueError('Undefined')
-            assert (isinstance(tmp, np.ma.masked_array))
+            assert (isinstance(tmp, np.ma.masked_array)), 'ERROR: wrong data type: ' + str(type(tmp))
```
[the remark this answers is not in the document]
Comment (bulli92) [reply geoval/core/data.py:2437]: ignore

File: geoval/core/data.py [hunk 10]
```diff
@@ -3549,7 +3547,7 @@ def _is_sorted(self):
         return np.all(np.diff(self.time) >= 0.)
 
 
-    def mask_region(self, r, return_object=False, method='full', maskfile=None, force=False):
+    def mask_region(self, r, return_object=True, method='full', maskfile=None, force=False):
```
[the remark this answers is not in the document]
Comment (bulli92) [reply geoval/core/data.py:3550]: only 1 empty line; delete remaining ones

File: setup_extensions.py [hunk 11]
```diff
@@ -11,11 +11,11 @@
 This is for development purposes only! Later on
 it might be integrated into the standard setup.py
 """
+from Cython.Build import cythonize
```
[the remark this answers is not in the document]
Comment (bulli92) [reply setup_extensions.py:14]: move import to top of file

File: tests/test_data.py [hunk 12]
```diff
@@ -543,7 +546,7 @@ def testTemporalTrendNoTimeNormalization(self):
         slope, intercept, r_value, p_value, std_err = stats.linregress(self.D.time,y)
 
         # calculate temporal correlation WITHOUT normalization of time
-        R, S, I, P = self.D.temporal_trend()  # no object is returned (default)
+        R, S, I, P = self.D.temporal_trend(return_object=False)  # no object is returned (default)
```
[the remark this answers is not in the document]
Comment (bulli92) [reply tests/test_data.py:549]: ignore

**Comment by bulli92** [comment]
The most critical aspect is here that there seem to be actual conflicts for data.py and setup_extensions.py

As no changes were made since a while, this is weird. Have you updated your local branch with `git pull ` before you started to implement your changes? Please solve the conflicts and fix also the stickler related issues like indicated.

**Comment by BenMGeo** [comment]
I think I had a pull request to master instead of dev, where I pulled from. 
I didn't make most of the changes. 

So I repull to dev?
Or make the changes to master and repull to master?

**Comment by bulli92** [comment]
repull to dev and not to master!


actually the right way would be to implement your changes in a branch 
that is based on dev and then make the PR to dev.

But I guess that's what you did (do changes within dev).

**Comment by bulli92** [comment]
not relevant any more

**Pull request closed without merging**